include/psp2/kernel/threadmgr.h

```diff
@@ -1228,6 +1228,146 @@ void *sceKernelGetThreadTLSAddr(SceUID thid, int key);
  */
 void *sceKernelGetTLSAddr(int key);
 
+/* RWLock */
+
+/** Additional options used when creating rwlock. */
+typedef struct SceKernelRWLockOptParam {
+	/** Size of the ::SceKernelRWLockOptParam structure */
+	SceSize     size;
+} SceKernelRWLockOptParam;
+
+/** Current state of a rwlock.
+ * @see sceKernelGetRWLockInfo
+ */
+typedef struct SceKernelRWLockInfo {
+	/** Size of the ::SceKernelRWLockInfo structure */
+	SceSize         size;
+	/** The UID of the rwlock */
+	SceUID          rwlockId;
+	/** NULL-terminated name of the rwlock */
+	char            name[32];
+	int             unk[5];
```

Comment:
Don't use non-fixed size types.
Also, that structure described on wiki: https://wiki.henkaku.xyz/vita/SceKernelThreadMgr